ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/MemcacheClient.java

```diff
@@ -31,15 +38,38 @@
  */
 public class MemcacheClient<K, V> implements TypedCache<K, V> {
   private static final long MAX_EXPIRATION_SEC = 60 * 60 * 24 * 30;
+  private static final String COMPONENT = MemcacheClient.class.getSimpleName();
 
   private final MemcachedClientIF spyClient;
   private final CacheKeyTranslator<K> keyTranslator;
   private final int expirationSpyUnits;
 
-  public MemcacheClient(final MemcachedClientIF spyClient, final CacheKeyTranslator<K> keyTranslator, final long expiration, final TimeUnit timeUnit) {
+  private MetricFactory metricFactory = null;
```

Comment:
I agree with Michael on this. Keeping the metric factory and relying on it to check other members is error prone. The minor "complexity" created is   cancelled by benefits of locality